aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -613,6 +613,31 @@ Tensor& mul_(Tensor& self, const Scalar& other) {
   return at::mul_out(self, wrapped_scalar_tensor(other), self); // redispatch!
 }
 
+void size_and_device_check(const Tensor& self, const Tensor& other) {
+  // if one of the self or other is a Scalar Tensor
+  if (!(self.sizes().size() == 0 || other.sizes().size() == 0)) {
+    TORCH_CHECK(self.sizes() == other.sizes());
+  }
+  TORCH_CHECK(self.device() == other.device());
```

Comment:
As mentioned offline, this is not true for zero dim Tensors.
I think it is ok here to be more relaxed than the true op if it is hard to replicate the exact same checks.